### PR TITLE
Initial support for Preact ⚛️

### PR DESCRIPTION
**Pull request by mgechev**
This PR introduces application detector and route parser for preact-router.

The limitations are similar to the one that the React parser has.

Didn't have a chance to test the actual prefetching. The next step would be to implement [this](https://github.com/mgechev/guess-js-react-demo) application in preact in order to verify that everything works as expected.

### CODE REVIEW

**Comment by mgechev** [comment]
👍 fixed the import

**Comment by addyosmani** [comment]
👍

File: packages/parser/src/preact/index.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  const extractRoute = (c: ts.Node) => {
+    if (c.kind !== ts.SyntaxKind.JsxElement && c.kind !== ts.SyntaxKind.JsxSelfClosingElement) {
+      return;
+    }
+    let el: ts.JsxSelfClosingElement | ts.JsxOpeningElement = (c as ts.JsxElement).openingElement;
+    if (c.kind === ts.SyntaxKind.JsxSelfClosingElement) {
+      el = c as ts.JsxSelfClosingElement;
+    }
+    const module: Partial<RoutingModule> = {
+      lazy: (el.tagName as ts.Identifier).text === 'AsyncRoute',
```
[the remark this answers is not in the document]
Comment (mgechev) [reply packages/parser/src/preact/index.ts]: Thanks for looking into this! I'll push updates in the next few days.

**Comment by mgechev** [comment]
@developit I think the parser now correctly recognizes lazy routes. Would you take a quick look to see if I didn't miss something?

**Comment by addyosmani** [comment]
Friendly ping if you get a chance @developit ✨

**Comment by mgechev** [comment]
Btw, @developit, what would be the right way to add this webpack plugin to my Preact CLI app? I saw there's a support for extensions. Probably something in this direction?

**Comment by addyosmani** [comment]
As an FYI Jason is currently taking part in JSConf in Berlin for the next few days and may be delayed getting back to this PR.

**Comment by mgechev** [comment]
@developit do you think it'll be easy to add the `GuessPlugin` to an existing Preact CLI app?

**Comment by addyosmani** [comment]
After chatting with Minko, we're going to go ahead with landing this one. @prateekbh, if you would be interested in trying out GuessPlugin with Preact CLI to check on how easy it would be to integrate it that would be amazing (I'm unsure if the best path there is a cli wrapper).

**Comment by developit** [comment]
Sorry for letting this slip! The update looks good. For Preact CLI, a plugin can be created that wraps up the Webpack plugin:

```js
import GuessPlugin from 'guess-webpack'

export default options => config => {
  config.plugins.push(
    new GuessPlugin(options)
  );
};
```

**Pull request merged**